lib/chef/knife/core/bootstrap_context.rb

```diff
@@ -40,7 +40,16 @@ def initialize(config, run_list, chef_config, secret = nil)
         end
 
         def bootstrap_environment
-          @chef_config[:environment] || '_default'
+          # Check for environment in CLI, first_boot_attributes, then in the
+          # config file. Use '_default' if no environment specified.
+
+          # TODO: @chef_config picks up @config's environment setting, even when
```

Comment:
I left it as a reminder for myself to ask a question, then I figured out my answer. I'll leave this as a comment (but remove the TODO).